components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/asyncoperation/handlers/AbstractVirtualNodeOperationHandler.kt

```diff
@@ -1,46 +1,84 @@
 package net.corda.virtualnode.write.db.impl.writer.asyncoperation.handlers
 
 import net.corda.data.virtualnode.VirtualNodeOperationStatus
+import net.corda.libs.virtualnode.datamodel.dto.VirtualNodeOperationDto
 import net.corda.libs.virtualnode.datamodel.dto.VirtualNodeOperationStateDto
-import net.corda.messaging.api.publisher.Publisher
-import net.corda.messaging.api.records.Record
-import net.corda.schema.Schemas
-import org.slf4j.Logger
+import net.corda.libs.virtualnode.datamodel.dto.VirtualNodeOperationType
+import net.corda.libs.virtualnode.datamodel.repository.VirtualNodeRepository
+import net.corda.libs.virtualnode.datamodel.repository.VirtualNodeRepositoryImpl
+import net.corda.orm.utils.transaction
 import java.time.Instant
+import javax.persistence.EntityManagerFactory
 
 internal abstract class AbstractVirtualNodeOperationHandler(
-    private val statusPublisher: Publisher,
-    private val logger: Logger
+    private val entityManagerFactory: EntityManagerFactory,
+    private val virtualNodeRepository: VirtualNodeRepository = VirtualNodeRepositoryImpl()
 ) {
-    protected fun publishStartProcessingStatus(requestId: String) {
-        publishStatusMessage(requestId, getAvroStatusObject(requestId, VirtualNodeOperationStateDto.IN_PROGRESS))
+    protected fun publishStartProcessingStatus(requestId: String, virtualNodeOperationType: VirtualNodeOperationType) {
```

Comment:
IMO using term "publish" is not quite right here, as we are not really publishing anything in a sense of broadcasting. Therefore, perhaps better:
```suggestion
    protected fun recordStartProcessingStatus(requestId: String, virtualNodeOperationType: VirtualNodeOperationType) {
```
and rename all the "publish" methods below.

Reply:
I have renamed these methods